src/gui/editors/PianoRoll.cpp

```diff
@@ -4257,7 +4266,14 @@ bool PianoRoll::toggleStepRecording()
 		}
 	}
 
-	return m_stepRecorder.isRecording();;
+	bool isRecording = m_stepRecorder.isRecording();
+
+	// hide playhead when step recording
+	m_timeLine->setPlayheadVisible(!isRecording);
+	m_positionLine->setVisible(!isRecording);
+	m_positionLine->update();
```

Comment:
Is this call to `update` needed? I would imagine `setVisible` handles that, but I'm not sure.